pyAKI/kdigo.py

```diff
@@ -40,37 +71,61 @@ def __init__(
                 ),
                 DemographicsPreProcessor(stay_identifier=stay_identifier),
             ]
-
+        # apply preprocessors to the input data
         for preprocessor in preprocessors:
             data = preprocessor.process(data)
 
-        self._data = data
-        self._probes = probes
-        self._stay_identifier = stay_identifier
+        self._data: list[Dataset] = data
+        self._probes: list[Probe] = probes
+        self._stay_identifier: str = stay_identifier
 
     def process_stays(self) -> pd.DataFrame:
+        """
+        Process all stays in the input data.
+
+        This method processes all stays in the input data by applying the configured probes and preprocessors.
+        The analysis results for all stays are concatenated and returned as a single DataFrame.
+
+        Returns:
+            pd.DataFrame: The analysis results for all stays.
+        """
         (_, df), *datasets = self._data
-        stay_ids = df.index.get_level_values("stay_id").unique()
+        stay_ids: pd.Index = df.index.get_level_values("stay_id").unique()
         for _, df in datasets:
             stay_ids.join(df.index.get_level_values("stay_id").unique())
 
-        data = self.process_stay(stay_ids.values[0])
+        data: pd.DataFrame = self.process_stay(stay_ids.values[0])
         for stay_id in stay_ids.values[1:]:
             data = pd.concat([data, self.process_stay(stay_id)])
 
         return data
 
     def process_stay(self, stay_id: str) -> pd.DataFrame:
+        """
+        Process a specific stay in the input data by patient identificator.
+
+        This method processes a specific stay in the input data by applying the configured probes and preprocessors.
+        The analysis results for the stay are returned as a DataFrame.
+
+        Args:
+            stay_id (str): The identifier of the stay to process.
+
+        Returns:
+            pd.DataFrame: The analysis results for the specific stay.
+        """
+
         data = [(name, data.loc[stay_id]) for name, data in self._data]
 
         for probe in self._probes:
-            data = probe.probe(data)
+            data: pd.DataFrame = probe.probe(data)
```

Comment:
```suggestion
            data: Dataset = probe.probe(data)
```
The type hint should be Dataset